tests/test_renderers_slate.py

```diff
@@ -0,0 +1,176 @@
+import os
+import pathlib
+import tempfile
+import shutil
+
+import pytest
+
+from lablib.lib import SequenceInfo
+from lablib.generators import SlateHtmlGenerator, SlateFillMode
+from lablib.renderers import SlateRenderer
+
+
+SLATE_TEMPLATE_FILE = os.path.join(
+    os.path.dirname(__file__),
+    "..",
+    "templates",
+    "slates",
+    "slate_generic",
+    "slate_generic.html"
+)
+DEFAULT_SEQUENCE_PATH = "resources/public/plateMain/v002"
+DEFAULT_SEQUENCE = SequenceInfo.scan(DEFAULT_SEQUENCE_PATH)[0]
+
+
+def _run_slate_renderer(
+        generator: SlateHtmlGenerator,
+        sequence: SequenceInfo = None,
+        output_path: str = None,
+    ):
+    """ Run a renderer associated with the provided
+        generator and source sequence.
+    """
+    sequence = sequence or DEFAULT_SEQUENCE
+    renderer = SlateRenderer(
+        generator,
+        sequence,
+        destination=output_path,
+    )
+
+    renderer.render()
+
+
+@pytest.fixture()
+def source_dir():
+    """ Prepare are clean source sequence
+        and after each tests.
+    """
+    # Prepare a temporary directory with a copy of default sequence
+    temp_dir = tempfile.mkdtemp(dir=os.getcwd())
+    for img in DEFAULT_SEQUENCE.frames:
+        new_path = pathlib.Path(temp_dir) / img.filename
+        shutil.copy(img.filepath, new_path)
+
+    yield temp_dir
+
+    # Remove all temporary directory content.
+    shutil.rmtree(temp_dir)
```

Comment:
Lets implement the fixture from `tests\conftest.py` for **test_staging_dir** please.

Reply:
Also it would be great for later test result analyses if each of test case would output into its own folder.

Reply:
Just to be sure this is what you had in mind.

I've adjusted this fixture to leverage the `conftest.py/test_staging_dir` one, for each test it will create a dedicated folder specific for this test and copy the resource sequence files over (so we can test the default prepend slate destination behavior).  At the end of the test session `test_staging_dir` is deleted which cleans everything.

This way we ensure each test result is scoped .